src/python/pants/backend/python/goals/coverage_py_integration_test.py

```diff
@@ -369,3 +369,48 @@ def test_1():
         )
         in result.stderr
     ), result.stderr
+
+
+def test_global_coverage() -> None:
+    files = {
+        "minimalcov/minimalcov/src/foo.py": 'print("In the foo module!")',
+        "minimalcov/minimalcov/src/foo_not_covered.py": 'print("No test coverage!")',
+        "minimalcov/minimalcov/src/BUILD": "python_sources()",
+        "minimalcov/minimalcov/tests/test_foo.py": dedent(
+            """\
+            import minimalcov.src.foo
+
+            def test_1():
+                assert True
+            """
+        ),
+        "minimalcov/minimalcov/tests/BUILD": "python_tests()",
+    }
+
+    # global coverage should include even those files for which there are no tests
+    with setup_tmpdir(files) as tmpdir:
+        command = [
+            "--backend-packages=pants.backend.python",
+            "test",
+            "--use-coverage",
+            "::",
+            f"--source-root-patterns=['/{tmpdir}/minimalcov']",
+            "--coverage-py-report=raw",
+            "--coverage-py-global-report",
```

Comment:
@jsirois I suppose this is how the global coverage option should be passed? I started writing the test to just check how this would even work. But it doesn't show the file for which there is no test in coverage report.

Reply:
It is. If you're ever in doubt, one way to check is `./pants help coverage-py`. Another is https://www.pantsbuild.org/docs/reference-coverage-py#global_report

Reply:
Can you help me out a bit more? What do you mean by "but not in the test"? Does Pants error or just produce result you do not expect. Either way can you provide full test output (`pants test --output=all ...` and what you expected instead?

Reply:
This test is pretty much what I am trying to do. As you can see there are two two python modules inside `minimalcov/minimalcov/src/` but only one of them has been imported in the test. With the global `--coverage-py-global-report` option enabled, I expected that the coverage report would include the module (`foo_not_covered.py`) which hasn't be imported in the test with 0% coverage. This is what I am trying to assert check in the test. Although the output only includes `foo.py`

```python
def test_global_coverage() -> None:
    files = {
        "minimalcov/minimalcov/src/foo.py": 'print("In the foo module!")',
        "minimalcov/minimalcov/src/foo_not_covered.py": 'print("No test coverage!")',
        "minimalcov/minimalcov/src/BUILD": "python_sources()",
        "minimalcov/minimalcov/tests/test_foo.py": dedent(
            """\
            import minimalcov.src.foo
            def test_1():
                assert True
            """
        ),
        "minimalcov/minimalcov/tests/BUILD": "python_tests()",
    }

    # global coverage should include even those files for which there are no tests
    with setup_tmpdir(files) as tmpdir:
        command = [
            "--backend-packages=pants.backend.python",
            "test",
            "--use-coverage",
            "::",
            f"--source-root-patterns=['/{tmpdir}/minimalcov']",
            "--coverage-py-report=raw",
            "--coverage-py-global-report",
        ]
        result = run_pants(command)
        result.assert_success()
    assert (
        dedent(
            f"""\
            Name                                                  Stmts   Miss  Cover
            -------------------------------------------------------------------------
            {tmpdir}/minimalcov/minimalcov/src/foo.py              1      0   100%
            {tmpdir}/minimalcov/minimalcov/src/foo_not_covered.py  1      1   0%
            {tmpdir}/minimalcov/minimalcov/tests/test_foo.py       3      0   100%
            -------------------------------------------------------------------------
            TOTAL                                                     5      1   80%
            """
        )
        in result.stderr
    ), result.stderr
```

Reply:
Hi @jsirois, if you would have some time to have a look, I would appreciate it. It's just I am bit out of ideas on what I am missing here.

Reply:
Sorry for the delay @ShantanuKumar, I'll have this looked at by tomorrow evening US / Pacific. There is a lot to page in here; so code review won't cut it. I'll need to check this out and play with coverage / Pants implementation of it to get this all back in my head.

Reply:
Thank you very much! I haven't actually added any new code. Just tried this test to see how such integration tests work.

Reply:
Ok, this was just bad source roots and then that caused you to work around that with bad imports! The whole source root thing continues to be something Pants is not getting across clearly. This fix gets the test green. I only changed the not covered file name due to trying to get the very titchy test text matching to work - the column alignments in the output are ... not awesome to work with. I know you were just sticking with the test convention here, but there has got to be a better way.